### ISSUE TITLE
What repositories need added to default config.json?

### ISSUE THREAD

**Issue by S0AndS0** [git-utilities/fix_logs#1]
To help @llSourcell fix Git logs, the current [`config.json`](https://github.com/git-utilities/fix_logs/blob/master/config.json) file contains entries similar to...

> Update notice, `57` repositories are now within the default `config.json` file the following example configurations are left as is for brevity.

```JSON
{
  "fixed": "./fixed.json",
  "failed": "./failed.json",
  "defaults": {
    "origin_branch": "master",
    "origin_remote": "origin",
    "source_branch": "master",
    "source_remote": "source",
    "fix_branch": "fix",
    "fix_commit": "Fixes logs",
    "keep_fix_branch": false,
    "no_push": false
  },
  "repos": [
    {
      "dir": "~/git/hub/llSourcell/Bitcoin_Trading_Bot",
      "source": "https://github.com/jaungiers/Multidimensional-LSTM-BitCoin-Time-Series.git"
    },
    {
      "dir": "~/git/hub/llSourcell/How-to-Predict-Stock-Prices-Easily-Demo",
      "source": "git@github.com:jaungiers/LSTM-Neural-Network-for-Time-Series-Prediction.git"
    },
    {
      "dir": "~/git/hub/llSourcell/How_to_simulate_a_self_driving_car",
      "source": "git@github.com:naokishibuya/car-behavioral-cloning.git"
    }
  ]
}
```

... there are likely more repositories that can be added to automated fixing of Git logs.

If you're willing to help track these repositories down then please either list them as a response to this Issue, or via a Pull Request to the default [`config.json`](https://github.com/git-utilities/fix_logs/blob/master/config.json) file.

When adding to this list it would be very kind if you could also notate any extra commands required.

___

**Example**

- After checkout [`miloharper/simple-neural-network`](https://github.com/miloharper/simple-neural-network) requires `git mv`

```Bash
git mv ./main.py ./demo.py
```

___

This will help with directing future feature developments of the `fix_logs` project; currently I'm contemplating adding options such as...

```JSON
{
  "dir": "~/git/hub/llSourcell/How_to_simulate_a_self_driving_car",
  "source": "git@github.com:naokishibuya/car-behavioral-cloning.git",
  "git_commands": {
    "after_checkout": ["mv", "./main.py", "./demo.py"]
  }
}
```

... to better facilitate fixing Git logs in an automated fashion.

**Comment by S0AndS0** [comment]
The [`config.json`](https://github.com/git-utilities/fix_logs/blob/master/config.json) file has been updated with repositories that could be easily parsed for source URLs. the following lists those that could be found that require _special_ attention and/or actions to correct.

___


**Missing Source Repository**


> If the source is found then please add it to either the `config.json` file via Pull Request, or by adding to this Issue


- how_to_deploy_a_keras_model_to_production `=>` https://github.com/moinudeen

- world_models `=>` https://github.com/AppliedDataSciencePartners

- YOLO_Object_Detection `=>` https://github.com/thtrieu

- Intro_to_the_Math_of_intelligence `=>` https://github.com/mattnedrich

- The_evolution_of_gradient_descent `=>` https://github.com/GRYE

- LSTM_Networks `=>` https://github.com/kevin-bruhwiler

- AI_Dresses_Itself `=>` https://github.com/ikostrikov

- Gaussian_Mixture_Models `=>` https://github.com/brianspiering

- Q_Learning_Explained `=>` https://medium.com/@m.alzantot

- Boston_Dynamics_Atlas_Explained `=>` https://github.com/Einsbon/

- linear_regression_live `=>` https://github.com/mattnedrich

- loss_functions_explained `=>` https://github.com/smsaurabhv

- k_means_clustering `=>` https://github.com/gleydson404

- How_to_make_a_text_summarizer `=>` https://github.com/udibr

- navigating_a_virtual_world_with_dynamic_programming `=>` https://github.com/root-ua

- AI_for_Scheduling `=>` https://www.youtube.com/watch?v=nvLJq4GnCg4&feature=youtu.be

- synthetic_gradients_explained `=>` https://iamtrask.github.io/2017/03/21/synthetic-gradients/

- ethereum_demo `=>` https://github.com/eshon

- proximal_policy_optimization `=>` https://learningai.io/projects/2017/07/28/ai-gym-workout.html

- actor_critic `=>` https://towardsdatascience.com/reinforcement-learning-w-keras-openai-actor-critic-models-f084612cfd69

- what_is_an_initial_coin_offering `=>` https://github.com/gustavoguimaraes

- OpenMined_demo `=>` https://github.com/OpenMined

- Stock_Market_Prediction `=>` https://github.com/dineshdaultani

- A_guide_to_coreML `=>` https://github.com/gkswamy98

- Evolutionary_Space_Invaders `=>` https://github.com/citiususc

- Latent_Dirichlet_Allocation `=>` https://github.com/morreene

- naive_bayes_classifier `=>` https://github.com/alanbuzdar

- hyperparameter_optimization_strategies `=>` https://github.com/davidlibland

- random_forests `=>` https://github.com/rushter

- Second_Order_Optimization_Newtons_Method `=>` https://github.com/dsalaj

- tensorflow_image_classifier `=>` https://github.com/xblaster

- Dimensionality_Reduction `=>` https://github.com/rasbt

- logistic_regression_newtons_method `=>` https://github.com/eecs445-f16

- The_Power_of_the_blockchain `=>` https://github.com/k5trismegistus

- How-to-Learn-from-Little-Data `=>` https://github.com/hmishra2250

- How_to_make_an_evolutionary_tetris_bot `=>` https://github.com/IdreesInc/

- pytorch_in_5_minutes https://github.com/jcjohnson

- how_to_generate_video `=>` https://github.com/platers

- Policy_Gradients_to_beat_Pong `=>` https://github.com/karpathy

- hardware_demo `=>` https://github.com/etanzapinsky

- how_to_win_slot_machines `=>` https://github.com/awjuliani

- Game-AI `=>` https://github.com/kihyunwon


------


**Unlicensed Source Repositories**


> By default most works are considered "All rights reserved" when unlicensed! GitHub ToS may read as though there is _room to wiggle_ in regards to those that Fork properly, however, it is probably best to investigate further. The [OpenSource StackExchange -- How does GitHub's "forking right" cope with an "All rights reserved" project](https://opensource.stackexchange.com/questions/1154/) question may be worth reading too.


- Classifying_Data_Using_a_Support_Vector_Machine `=>` https://github.com/MaviccPRP/svm/blob/master/svm-primal.ipynb

- Convolutional_neural_network `=>` https://github.com/greydanus/pythonic_ocr

- How_to_Build_a_healthcare_startup `=>` https://github.com/shaqian/flutter_realtime_detection

- A_Guide_to_Running_Tensorflow_Models_on_Android `=>` https://github.com/mari-linhares/mnist-android-tensorflow

- Bitcoin_Trading_Bot `=>` https://github.com/jaungiers/Multidimensional-LSTM-BitCoin-Time-Series

- Pokemon_GAN `=>` https://github.com/moxiegushi/pokeGAN

- Drone_Flight_Controller `=>` https://github.com/sksq96/deep-rl-quadcopter

- Simple_Blockchain_in_5_Minutes `=>` https://github.com/howCodeORG/Simple-Python-Blockchain


------


**Requires Manual Corrections**


> Some are non-GitHub sources, others are from Gists, all **cannot** be automatically merged via this project currently.


- How_to_make_a_chatbot `=>` https://github.com/fchollet/keras/blob/master/examples/babi_memnn.py

- image_classifier_template `=>`

  - https://github.com/awesomedata/awesome-public-datasets
  - https://toolbox.google.com/datasetsearch

- recurrent_neural_network `=>` https://gist.github.com/karpathy/d4dee566867f8291f086

- Data_Visualization `=>` https://github.com/WillKoehrsen/Data-Analysis/blob/master/plotly/Plotly%20Whirlwind%20Introduction.ipynb

  - Note, Data_Visualization is missing the MIT license

- Time_Series_Prediction `=>` https://github.com/anirudhpillai16/Time-Series-Analysis/blob/master/Time%20Series%20Analytics%20Vidhya.ipynb

- how_to_use_tensorboard_live `=>` https://github.com/mamcgrath/TensorBoard-TF-Dev-Summit-Tutorial/blob/master/mnist.py

  - Note, how_to_use_tensorboard_live is missing Apache-2.0 license

- Internet_of_Things_Optimization https://github.com/dennybritz/reinforcement-learning

  - Note, Internet_of_Things_Optimization is missing the MIT license

- serverless_computing_with_google_Cloud `=>` https://github.com/ZhengzhengYu/tensorflow/blob/master/Wide%20and%20deep%20model%20for%20income%20classification.ipynb

- Your_First_Decentralized_Application `=>` https://gist.github.com/maheshmurthy

- math_of_machine_learning `=>` https://github.com/Shreyas3108/house-price-prediction/blob/master/housesales.ipynb

- AI_for_video_games_demo `=>` https://gist.github.com/malzantot

- Predicting_Winning_Teams `=>` https://github.com/RudrakshTuwani/Football-Data-Analysis-and-Prediction/blob/master/Prediction/Scraping%20and%20Cleaning.ipynb

- neural_networks `=>` http://iamtrask.github.io/2015/07/12/basic-python-network/

- how_to_generate_images `=>` https://github.com/fchollet/keras/blob/master/examples/variational_autoencoder.py


------


**Second Opinion Needed... Maybe**


> Not certain how to classify these, some or all may be okay as is, but probably best that their double checked to ensure that nothing is amiss


- Financial_Forecasting_with_TensorflowJS

  1. [**TensorFlow's** Official CNN Article!](https://www.tensorflow.org/tutorials/deep_cnn)
  2. [**Firdaouss Doukkali's** great introductory article!](https://medium.com/@phidaouss/convolutional-neural-networks-cnn-or-convnets-d7c688b0a207)
  3. [Special thanks to **Zaid Alyafeai**](https://medium.com/tensorflow/a-gentle-introduction-to-tensorflow-js-dba2e5257702) for several crystal-clear code samples!
  4. https://github.com/Thoughtscript/x_team_tensorflow_js

- sirajcoin

  1. https://github.com/mappum
  2. https://github.com/chadlohrli
  3. https://github.com/keppel

- neural_networks

  - [Install Conda](https://conda.io/docs/installation.html)

- Which Activation Function Should I Use `=>` https://codereview.stackexchange.com/questions/132023/different-neural-network-activation-functions-and-gradient-descent


------


**Missing ReadMe**


> Sources, if any will require searching for.


- [azure_machine_learning](https://github.com/llSourcell/azure_machine_learning)

- [new_repository](https://github.com/llSourcell/new_repository)

- [IPM-Quickstart-Android](https://github.com/llSourcell/IPM-Quickstart-Android)

- [HTTP-IPFS-Add](https://github.com/llSourcell/HTTP-IPFS-Add)

- [dapp](https://github.com/llSourcell/dapp)

- [kerala_test](https://github.com/llSourcell/kerala_test)

- [go-live-video-collage](https://github.com/llSourcell/go-live-video-collage)

- [Azadi_Complete](https://github.com/llSourcell/Azadi_Complete)

- [Satoshi](https://github.com/llSourcell/Satoshi)

- [Azadi](https://github.com/llSourcell/Azadi)

- [MotionSegment](https://github.com/llSourcell/MotionSegment)

- [FeatureClassifier](https://github.com/llSourcell/FeatureClassifier)

- [RetailRepunch2](https://github.com/llSourcell/RetailRepunch2)

- [LuaTest](https://github.com/llSourcell/LuaTest)

- [Blabber](https://github.com/llSourcell/Blabber)

- [BlabSay_Mobile_App](https://github.com/llSourcell/BlabSay_Mobile_App)


------


**Requires re-Parsing**


> Seems that three repositories where missed on _first-pass_ when searching for ReadMe files.


- [Sincerely](https://github.com/llSourcell/Sincerely)

- [Pano](https://github.com/llSourcell/Pano)

- [PlanSegmentation](https://github.com/llSourcell/PlanSegmentation)

**Comment by llSourcell** [comment]
hey, thanks so much for taking the time to compile this. If i can easily fix all my repos with a few commands to properly credit the authors, I will readily do that. The problem is if i have to manually go through all of them, that's going to be so much work. Can you please create a step by step tutorial on what i should do via command line to make that happen?

**Comment by S0AndS0** [comment]
> **@llSourcell**
> hey, thanks so much for taking the time to compile this. If i can easily fix all my repos with a few commands to properly credit the authors, I will readily do that. The problem is if i have to manually go through all of them, that's going to be so much work. Can you please create a step by step tutorial on what i should do via command line to make that happen?


___


Welcome as always.


Currently this project is focused on correcting things that if left as is could lead to [DMCA takedowns](https://www.dmca.com/FAQ/What-is-a-DMCA-Takedown) being issued, or worse. Reinstating original licenses and _injecting_ commit history from original authors seems to be the simplest way of avoiding such futures.


Not every repository explicitly states that accreditation is a requirement. So language processing may be able to help with marking repositories that require that any further credits, however, with my level of Machine Learning ignorance I'd likely resort to writing in-eloquent regex search patterns. Once marked though I'd be happy to help with writing a script and/or template that can append information to ReadMe files in an automated fashion.


The main ReadMe file of this project has a [Quick Start](https://github.com/git-utilities/fix_logs#quick-start) section that in the meantime may be of use, but I'll be certain to write a more detailed guide in the near future. What is the best way to notify you when it's ready for review?
